include/nbl/builtin/hlsl/tgmath/impl.hlsl

```diff
@@ -240,7 +247,13 @@ struct beta_helper<T NBL_PARTIAL_REQ_BOT(concepts::FloatingPointScalar<T>) >
 	// implementation from Numerical Recipes in C, 2nd ed.
 	static T __call(T v1, T v2)
 	{
-		return exp_helper<T>::__call(lgamma_helper<T>::__call(v1) + lgamma_helper<T>::__call(v2) - lgamma_helper<T>::__call(v1+v2));
+		// specialized for Cook Torrance BTDFs
+		assert(v1 >= 1.0 && v2 >= 1.0);
+
+		if (v1+v2 > 1e6)
+			return T(0.0);
```

Comment:
wrong header for this, you need a different `beta` in a different namespace for this

P.S. comment about the threshold, also threshold is different for each float size, you're looking for values that make the beta value exp2(-11), -24 and -53